File: appli-web/src/app/core/api/health-step-count-api.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { HealthStepCount } from '@core/models/health-step-count.model';
import { API_BASE_URL } from './api.config';

/**
 * Accès REST aux pas santé — **lecture seule côté web** : le web ne fait qu'afficher
 * ce que Health Connect (Android) a poussé, il n'écrit jamais ces buckets (pas d'upsert/delete).
 */
@Injectable({ providedIn: 'root' })
export class HealthStepCountApi {
  private readonly http = inject(HttpClient);
  private readonly base = `${API_BASE_URL}/health-step-counts`;

  getAll(): Observable<HealthStepCount[]> {
    return this.http.get<HealthStepCount[]>(this.base);
  }
}

File: appli-android/app/src/main/java/com/example/sportapp/core/network/HealthStepCountApi.kt
package com.example.sportapp.core.network

import com.example.sportapp.core.data.model.HealthStepCount
import retrofit2.http.*

interface HealthStepCountApi {

    @GET("health-step-counts")
    suspend fun getAll(): List<HealthStepCount>

    @GET("health-step-counts/{uuid}")
    suspend fun getByUUID(@Path("uuid") uuid: String): HealthStepCount

    @PUT("health-step-counts/{uuid}")
    suspend fun upsert(@Path("uuid") uuid: String, @Body item: HealthStepCount)

    @PUT("health-step-counts/bulk")
    suspend fun upsertAll(@Body items: List<HealthStepCount>)

    @DELETE("health-step-counts/{uuid}")
    suspend fun delete(@Path("uuid") uuid: String)
}
